code_search/kubeflow/components/nms.libsonnet

```diff
@@ -2,7 +2,49 @@ local baseParams = std.extVar("__ksonnet/params").components["nmslib"];
 
 {
 
-  nmsContainer(params, env):: {
+  indexCreator(params, env):: {
+    apiVersion: "v1",
+    kind: "Pod",
```

Comment:
Why are you using a Pod? You should always use a controller e.g. Deployment. A Pod will not be recreated if it is terminated. But a controller like a Deployment will create a new pod if the existing pod stops running.

Reply:
Makes sense. I did try using the "Job" kind but was running into certain issues. Perhaps some spec keys from "Deployment" kind are not compatible with "Job". I'll fix this.